airflow/models/dag.py

```diff
@@ -591,6 +593,29 @@ def owner(self) -> str:
         """
         return ", ".join({t.owner for t in self.tasks})
 
+    def code(self):
+        if conf.getboolean('core', 'store_dag_code', fallback=False):
+            return self._get_code_from_db()
+        else:
+            return self._get_code_from_file()
+
+    def _get_code_from_file(self):
+        with open_maybe_zipped(self.fileloc, 'r') as f:
+            code = f.read()
+        return code
+
+    @provide_session
+    def _get_code_from_db(self, session=None):
+        fileloc_hash = DagCode.dag_fileloc_hash(self.fileloc)
```

Comment:
I have moved them there, although I have doubts because what does method _get_code_from_file(self) does there.
But cyclic dependencies seem like a good reason to move.

Reply:
We should move it to another place, but Airflow ... has problematic architecture for now. It will be harder to change so that the database objects do not contain logic. Currently, very often they have very complex functions, in particular SchedulerJob, but ... this is a separate topic for discussion.

Reply:
Maybe we should introduce a new class DagCodeBag or something similar?

Reply:
In this case it seems to me that better would be enemy of good enough :)
There is no good place in current architecture. I agree that this solution is better than previous one. And it seems to me that DagBag will definitely gain functionality and this once proven to work will not be changed anymore.

Reply:
Done